idaes/power_generation/carbon_capture/compression_system/compressor.py

```diff
@@ -0,0 +1,659 @@
+        self.eff_p_v = Var(self.flowsheet().config.time,
+                           initialize=0.85,
+                           doc="Polytropic Efficiency Vaned Diffuser",
+                           bounds=(0.001, 1))
+        self.eff_p = Var(self.flowsheet().config.time,
+                         initialize=0.85,
+                         doc="Polytropic Efficiency",
+                         bounds=(0.001, 1.0))
+        self.Ma = Var(self.flowsheet().config.time,
+                      initialize=0.1,
+                      doc='Rotational Mach Number',
+                      bounds=(0, 5))
+        self.kappaT = Var(self.flowsheet().config.time,
+                          initialize=5.0e-7, units=1 / pyunits.Pa)
+        self.Ma_max = Var(self.flowsheet().config.time,
+                          initialize=0.1,
+                          doc='Rotational Mach Number Upper limit',
+                          bounds=(0, 5))
+        # dynamic related variables
+        self.psi_3 = Var(self.flowsheet().config.time,
+                         initialize=0.01,
+                         doc="Dimensionless Exit Flow Coefficient",
+                         within=NonNegativeReals,
+                         units=pyunits.dimensionless)
+        self.psi_s = Var(self.flowsheet().config.time,
+                         initialize=0.01,
+                         doc="Dimensionless Isentropic Head Coefficient",
+                         within=NonNegativeReals,
+                         units=pyunits.dimensionless)
+        ########################################################
+        # Declare parameters for the model
+        self.k_v = Param(default=1.2766,
+                         mutable=False,
+                         doc="Heat Capacity Ratio",
+                         within=NonNegativeReals)
+        self.z_s = Param(default=0.97373,
+                         mutable=True,
+                         doc="Compressibility factors at suction",
+                         within=NonNegativeReals)
+        self.z_d1 = Param(default=0.88949,
+                          mutable=True,
+                          within=NonNegativeReals)
+        self.dp = Param(default=1,
+                        within=NonNegativeReals,
+                        doc='A Pressure Difference'
+                        'for Finite Difference Derivative Calculation',
+                        units=pyunits.Pa)
+        self.efficiency_mech = Param(default=0.97,
+                                     doc="Mechanical Efficiency")
+        self.eff_drive = Param(default=1.0,
+                               doc="Driver efficiency")
+
+        ############################################################
+
+        self.ratioP[:] = 2.0   # make sure these have a number value
+        self.deltaP[:] = 0     # to avoid an error later in initialize
+
+        ###########################################################
+        # Speed of sound calculations
```

Comment:
If you test this with CO2 we can get the speed of sound from the property package.  I think we can just make speed of sound a requirement that this model has of property packages  that will get you out of needing a finite difference here and move this to the prop pack where it can be handled more appropriately.

Reply:
I second that - speed of sound is a thermophysical property, and so should be calculated by the property package.